### PR TITLE
feat(deep-link): implement getCurrent on Windows/Linux

### PR DESCRIPTION
**Pull request by lucasfernog**
checks std::env::args() on initialization, also includes integration with the single-instance plugin

### CODE REVIEW

File: plugins/deep-link/examples/app/src-tauri/src/lib.rs [hunk 1]
```diff
@@ -13,13 +13,41 @@ fn greet(name: &str) -> String {
...
+                    .plugins
+                    .0
+                    .get("deep-link")
+                    .unwrap()
+                    .get("desktop")
+                    .unwrap()
+                    .get("schemes")
+                    .unwrap()
+                    .as_array()
+                    .unwrap();
```
Comment (amrbashir) [inline plugins/deep-link/examples/app/src-tauri/src/lib.rs]: should we include a helper method on `DeepLinkExt` ? maybe the whole logic actually `app.deep_link().register_desktop()`

File: plugins/deep-link/src/lib.rs [hunk 2]
```diff
@@ -154,23 +166,55 @@ mod imp {
...
         pub(crate) app: AppHandle<R>,
-        #[allow(dead_code)]
         pub(crate) current: Mutex<Option<Vec<url::Url>>>,
+        pub(crate) config: Option<crate::config::Config>,
     }
 
     impl<R: Runtime> DeepLink<R> {
+        #[cfg(desktop)]
+        #[doc(hidden)]
+        pub fn handle_cli_arguments<S: AsRef<str>, I: Iterator<Item = S>>(&self, mut args: I) {
```
Comment (amrbashir) [inline plugins/deep-link/src/lib.rs]: if this is used by tauri-plugin-single-instance, maybe we should document it instead

Comment (lucasfernog) [reply plugins/deep-link/src/lib.rs]: should only be used internally tbh.. but i agree we should at least comment what it does, on it

Comment (amrbashir) [reply plugins/deep-link/src/lib.rs]: it is internal to our workspace yes, but if it is used in another plugin, it should be documented and potentiolly allow other users to add same integration

File: plugins/deep-link/examples/app/src-tauri/src/lib.rs (hunk 1 above)
Comment (lucasfernog) [reply plugins/deep-link/examples/app/src-tauri/src/lib.rs]: excellent idea i love you, pushed

File: plugins/deep-link/src/lib.rs [hunk 3]
```diff
@@ -154,23 +172,87 @@ mod imp {
...
         pub fn get_current(&self) -> crate::Result<Option<Vec<url::Url>>> {
-            #[cfg(not(any(windows, target_os = "linux")))]
             return Ok(self.current.lock().unwrap().clone());
-            #[cfg(any(windows, target_os = "linux"))]
-            Err(crate::Error::UnsupportedPlatform)
+        }
+
+        /// Registers all schemes defined in the configuration file.
+        ///
+        /// This is useful to ensure the schemes are registered even if the user unregistered it
```
Comment (amrbashir) [inline plugins/deep-link/src/lib.rs]: This feels wrong tbh and we shouldn't recommend that at all. If a user unregistered a protocol they should be in control of that and we shouldn't keep annoying them.

**Review by amrbashir — DISMISSED** [review]

File: plugins/deep-link/src/lib.rs (hunk 3 above)
Comment (amrbashir) [reply plugins/deep-link/src/lib.rs]: I know that app images and portable builds on Windows need to manually register that but we either warn against this or provide some checks that user chose to disable this protocol, so you shouldn't re-register them

Comment (lucasfernog) [reply plugins/deep-link/src/lib.rs]: oops oops oops :joy:

**Review by amrbashir — APPROVED** [review]

**Pull request merged**